orchagent/routeorch.cpp

```diff
@@ -2574,3 +2591,22 @@ void RouteOrch::decNhgRefCount(const std::string &nhg_index)
         gCbfNhgOrch->decNhgRefCount(nhg_index);
     }
 }
+
+void RouteOrch::publishRouteState(const RouteBulkContext& ctx, const ReturnCode& status)
+{
+    SWSS_LOG_ENTER();
+
+    std::vector<FieldValueTuple> fvs;
+
+    /* If the operation type is "DEL" then ctx.protocol is empty and fvs is left empty.
+     * An empty fvs makes ResponsePublisher::publish() remove the state entry from APPL_STATE_DB
+     */
+    if (!ctx.protocol.empty())
```

Comment:
Why introduce this field for identifying operation? I'm not sure if i understand the logic here. Can you please explaing this?

Reply:
It is not the motivation for introducing this field. The context does not carry operation type, however, if the protocol is left empty that means it is a delete operation, since the delete update does not carry FVS (including protocol)

Reply:
@prsunny Replaced with a flag saved in the context object, please check.